src/dragger.js

```diff
@@ -0,0 +1,126 @@
+ * be moved while someone is dragging it.
+ */
+export class MultiSelectBlockDragger extends Blockly.BlockDragger {
+  /** @override */
+  constructor(block, workspace) {
+    super(block, workspace);
+    this.block_ = block;
+    this.workspace_ = workspace;
+    this.group_ = '';
+    this.blockDraggers_ = new Set();
+  }
+
+  /**
+   * Prepares the block dragger for a new drag.
+   * @param {!Blockly.utils.Coordinate} currentDragDeltaXY How far the pointer
+   *     has moved from the position at the start of the drag, in pixel units.
+   * @param {boolean} healStack Whether or not to heal the stack after
+   *     disconnecting.
+   */
+  startDrag(currentDragDeltaXY, healStack) {
+    if (!inMultipleSelectionMode) {
+      const blockDraggerList = [];
+      if (!blockSelection.has(this.block_.id)) {
+        blockSelection.forEach((id) => {
+          const element = this.workspace_.getBlockById(id);
+          if (!element.disposed) {
+            element.pathObject.updateSelected(false);
+          }
+        });
+        blockSelection.clear();
+        blockDraggerList.push(new Blockly.BlockDragger(this.block_,
+            this.workspace_));
+        this.block_.pathObject.updateSelected(true);
+      }
+      blockSelection.forEach((id) => {
+        const element = this.workspace_.getBlockById(id);
+        if (element.disposed) {
+          blockSelection.delete(id);
+          return;
+        }
+        // Only drag the parent if it is selected.
+        if (hasSelectedParent(element)) {
+          return;
+        }
+        blockDraggerList.push(new Blockly.BlockDragger(element,
+            this.workspace_));
+      });
+      blockDraggerList.forEach((blockDragger) => {
+        blockDragger.startDrag(currentDragDeltaXY, healStack);
+        this.blockDraggers_.add(blockDragger);
+      });
+      if (this.blockDraggers_.size > 1) {
+        // Disabled the highlighting around connection for multiple blocks
+        // dragging because of the bugs.
+        Blockly.RenderedConnection.prototype.highlight = function() {};
+      }
+    }
+  }
+
+  /**
+   * Moves the block to the specified location.
+   * @param {!Event} e The mouseup/touchend event.
+   * @param {!Blockly.utils.Coordinate} currentDragDeltaXY How far the pointer
+   *     has moved from the position at the start of the drag, in pixel units.
+   */
+  drag(e, currentDragDeltaXY) {
+    this.blockDraggers_.forEach(function(blockDragger_) {
+      blockDragger_.drag(e, currentDragDeltaXY);
+    });
+    e.preventDefault();
+    e.stopPropagation();
+  }
+
+  /**
+   * Finishes the block drag.
+   * @param {!Event} e The mouseup/touchend event.
+   * @param {!Blockly.utils.Coordinate} currentDragDeltaXY How far the pointer
+   *     has moved from the position at the start of the drag, in pixel units.
+   */
+  endDrag(e, currentDragDeltaXY) {
+    this.blockDraggers_.forEach((blockDragger_) => {
+      if (Blockly.Events.getGroup()) {
```

Comment:
Have you tested this with undo and redo? From my reading I think you're missing a `setGroup` call to make sure all of the drags can be undone/redone together. But there's also the possibility the individual block draggers will trample over your grouping anyway...

If it isn't working correctly then we can work from there haha.

Reply:
Yeah, I have tested that with undo and redo, all of the drags can be undone/redone together.

This is actually a intentionally hack, as `startDrag` https://github.com/google/blockly/blob/master/core/block_dragger.js#L129-L131 and `endDrag` 
https://github.com/google/blockly/blob/master/core/block_dragger.js#L287 set group id separately, what I'm doing here is to modify the group id before it goes into the stack so that those individual block draggers can share the same group id. Without these code, the group id will be empty in a multiple drag.

`Blockly.getMainWorkspace().getUndoStack()`:
![image](https://user-images.githubusercontent.com/43995067/178641279-2d774b3b-af1d-4660-a292-1a5d4e108fc9.png)

Reply:
Awesome! Thanks for testing =)